### PR TITLE
Conda build changes

### PR DESCRIPTION
**Pull request by sfc-gh-chu**
# Description
Changes around enabling building and packaging to snowflake conda channel. 
 - removes dependencies to `humanize`, `merkle-json`, `tqdm`, and `millify`.
 - Fixes missing requirements and versioning to satisfy conda constraints
 - Adds source distribution to pypi release

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/core/trulens/core/tru.py [hunk 1]
```diff
@@ -57,10 +55,6 @@
 logger = logging.getLogger(__name__)
 
 
-def humanize_seconds(seconds: float):
-    return humanize.naturaldelta(timedelta(seconds=seconds))
```
[the remark this answers is not in the document]
Comment (sfc-gh-pmardziel) [reply src/core/trulens/core/tru.py:61]: Ohh, nm, I see the commented out usage did not call the replacement and just printed "seconds". Will add note below.

File: src/core/trulens/core/feedback/feedback.py [hunk 2]
```diff
@@ -408,7 +408,7 @@ def prepare_feedback(
 
             # TODO: figure out useful things to print.
             # feedback_ident = (
-            #     f"[last seen {humanize.naturaldelta(elapsed)} ago] "
+            #     f"[last seen {elapsed} seconds ago] "
```
Comment (sfc-gh-pmardziel) [inline src/core/trulens/core/feedback/feedback.py:411]: Use the replaced method to format elapsed?

File: src/core/trulens/core/tru.py [hunk 3]
```diff
@@ -881,51 +873,15 @@ def runloop():
...
             )
             print(
                 f"Will rerun failed feedbacks after "
-                f"{humanize_seconds(self.RETRY_FAILED_SECONDS)}."
+                f"{format_seconds(self.RETRY_FAILED_SECONDS)}."
             )
 
             total = 0
 
-            # Getting total counts from the database to start off the tqdm
```
Comment (sfc-gh-pmardziel) [inline src/core/trulens/core/tru.py:893]: Why were the progress bars removed?

File: src/core/trulens/core/utils/json.py [hunk 4]
```diff
@@ -52,10 +61,45 @@ def encode_openai_timeout(obj: Timeout) -> Dict[str, Any]:
...
+
+            acc += f"{k}:{_recursive_hash(key_val, ignore_none=ignore_none)},"
+        return _recursive_hash(acc, ignore_none=ignore_none)
+    elif isinstance(
+        value,
+        (str, int, bool, float, complex),
+    ):
+        return hashlib.md5(str(value).encode("utf-8")).hexdigest()
+    elif value is None:
+        return _recursive_hash(str(value), ignore_none=ignore_none)
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/utils/json.py:99]: This is base case right? why send this again in recrursive-hash?

File: src/core/trulens/core/utils/json.py [hunk 5]
```diff
@@ -52,10 +61,45 @@ def encode_openai_timeout(obj: Timeout) -> Dict[str, Any]:
 
 logger = logging.getLogger(__name__)
 pp = PrettyPrinter()
-
 T = TypeVar("T")
 
-mj = MerkleJson()
+
+def _recursive_hash(
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/utils/json.py:67]: Should we write a UTs for this method? Or the public method using this?

File: src/core/trulens/core/tru.py [hunk 6]
```diff
@@ -881,51 +873,15 @@ def runloop():
...
-                f"{humanize_seconds(self.RETRY_RUNNING_SECONDS)}."
+                f"{format_seconds(self.RETRY_RUNNING_SECONDS)}."
             )
             print(
                 f"Will rerun failed feedbacks after "
-                f"{humanize_seconds(self.RETRY_FAILED_SECONDS)}."
+                f"{format_seconds(self.RETRY_FAILED_SECONDS)}."
             )
 
             total = 0
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/tru.py:883]: What's the reason for removing the tqdm stats?

**Review by sfc-gh-pdharmana — COMMENTED** [review]
We need UTs for all the new functions implemented.

File: src/core/trulens/core/tru.py (hunk 6 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/tru.py:883]: Was working on stripping out non-essential dependencies from trulens-core but can bring back the stats if needed

File: src/core/trulens/core/tru.py (hunk 3 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/tru.py:893]: From another thread

> Was working on stripping out non-essential dependencies from trulens-core but can bring back the stats if needed

File: src/core/trulens/core/schema/app.py [hunk 7]
```diff
@@ -121,8 +120,6 @@ def __init__(
                     logger.warning(
                         "`initial_app_loader` dump is too big (%s) > %s bytes). "
                         "If you are loading large objects, include the loading logic inside `initial_app_loader`.",
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/schema/app.py:122]: may be print the dump trimmed to a certain length?

File: src/core/trulens/core/tru.py (hunk 6 above)
Comment (sfc-gh-pdharmana) [reply src/core/trulens/core/tru.py:883]: May be file another JIRA for the stats.

File: tests/unit/core/utils/test_json_utils.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+@pytest.mark.parametrize(
+    "test_input,matching_input,expected",
+    [
+        ("", None, STR_HASH := "d41d8cd98f00b204e9800998ecf8427e"),
+        ("hello", None, STR_HASH := "5d41402abc4b2a76b9719d911017c592"),
+        (5, "5", INT_HASH := "e4da3b7fbbce2345d7772b0674a318d5"),
+        (3.4, "3.4", FLOAT_HASH := "31053ad0506e935470ca21b43cae98cf"),
+        (True, "True", BOOL_HASH := "f827cf462f62848df37c5e1e94a4da74"),
+        (None, "None", NONE_HASH := "6adf97f83acf6453d4a6a4b1070f3754"),
```
Comment (sfc-gh-pdharmana) [inline tests/unit/core/utils/test_json_utils.py:13]: There is `complex` data type in the hash, do we need a test case for that?

File: Makefile [hunk 9]
```diff
@@ -123,7 +123,7 @@ clean: clean-dashboard
 build-dashboard: env clean-dashboard
 	poetry run python -m build src/dashboard -o $(REPO_ROOT)/dist/trulens-dashboard;
 
-build: $(POETRY_DIRS)
+build: $(POETRY_DIRS) lock
```
Comment (sfc-gh-pdharmana) [inline Makefile:126]: why do we need this lock?

**Review by sfc-gh-pdharmana — APPROVED** [review]
Looks good overall, we probably need a JIRA for an alternate progress library changes.

**Pull request merged**